tools/define_sanity/check.py

```diff
@@ -0,0 +1,93 @@
+import fnmatch
+import glob
+import os
+import re
+import sys
+
+parent_directory = "code/**/*.dm"
+
+output_file_name = "define_sanity_output.txt"
+how_to_fix_message = "Please #undef the above defines or remake them as global defines in the code/__DEFINES directory."
+
+def green(text):
+    return "\033[32m" + str(text) + "\033[0m"
+
+def red(text):
+    return "\033[31m" + str(text) + "\033[0m"
+
+def blue(text):
+    return "\033[34m" + str(text) + "\033[0m"
+
+def post_error(define_name, file, github_error_style):
+    if github_error_style:
+        print(f"::error file={file},title=Define Sanity::{define_name} is defined locally in {file} but not undefined locally!")
+    else:
+        print(red(f"- Failure: {define_name} is defined locally in {file} but not undefined locally!"))
+
+# simple way to check if we're running on github actions, or on a local machine
+on_github = os.getenv("GITHUB_ACTIONS") == "true"
+
+# This files/directories are expected to have "global" defines, so they must be exempt from this check.
+# Add directories as string here to automatically be exempt in case you have a non-complaint file name.
+excluded_files = [
+    #  Wildcard directories, all files are expected to be exempt.
+    "code/__DEFINES/*.dm",
+    "code/__HELPERS/*.dm",
```

Comment:
This is a later project, but __HELPERS having defines has bitten us a decent amount and so one day I would like to remove this exception